calculator.c

```diff
@@ -0,0 +1,77 @@
+#include<stdio.h>
+#include<stdlib.h>
+#include<ctype.h>
+#include<string.h>
+void removewhitespacesfromexpression(char input[]) {
+    int start = 0, start1 = 0;
+    while (input[start] != '\0') {
+        if (!isspace(input[start])) {
+            input[start1] = input[start]; 
+            start1++;
+        }
+        start++;
+    }
+    input[start1] = '\0';
+}
+
+int maincalculation(int op1, int op2, char op) {
+    if (op == '+') return op1 + op2; 
+    if (op == '-') return op1 - op2;
+    if (op == '*') return op1 * op2; 
+    if (op == '/') {
+        if (op2 != 0) {
+            return op1 / op2; 
+        }else{
+        printf("Error:Division by zero.\n");
+        exit(1); 
+    }
+    }
+}
+
+int mainoperation(char input[]) {
+    int startnumber = 0, endnumber = 0, ans = 0;
+    char op = 'x';
+    int i = 0;
+    while (isdigit(input[i])) {
+        startnumber = startnumber * 10 + (input[i] - '0');
+        i++;
+    }
+    while (input[i] != '\0') {
+        if (strchr("*/+-",input[i])) {
+            op = input[i];
+            i++;
+        } else {
+            printf("Invalid expression.\n");
+            return 0;
+        }
+        endnumber = 0;
+        while (isdigit(input[i])) {

```

Comment:
Avoid using inbuilt functions unless it is system calls, please change the built-in functions to user implementation function. take care at all places.